envisage/plugins/remote_editor/communication/util.py

```diff
@@ -20,10 +24,25 @@
                          'remote_editor_server.lock')
 LOG_PATH = os.path.join(ETSConfig.application_data, 'remote_editor_server.log')
 
+# Python 3 compatibility
+PY3 = sys.version_info[0] >= 3
+STRING_BASE_CLASS = str if PY3 else basestring
+
+if PY3:
+    def encode(s):
+        return bytes(s, 'utf-8')
+    def decode(s):
+        return s.decode()
```

Comment:
I suspect this will need a unicode-based test to ensure that it does actually work in both Python 2 and Python 3.